chapter1.Rmd

```diff
@@ -462,17 +547,23 @@ class(my_logical)
+# ยินดีด้วย! คุณได้ทำแบบฝึกหัดของบทนี้ครบทั้งหมดแล้ว! บทต่อไปเราจะได้เรียนรู้มากขึ้นเกี่ยวกับสิ่งที่เรียกว่า vectors!
+
 lapply(c("my_numeric", "my_character", "my_logical"), test_object, undefined_msg = msg, incorrect_msg = msg)
-patt <- "Have you included `class(%1$s)` to print out the data type of `%1$s`?"
-test_output_contains("class(my_numeric)",
-                     incorrect_msg = "Do not remove the code that prints out the type of `my_numeric`.")
-test_output_contains("class(my_character)",
-                     incorrect_msg = sprintf(patt, "my_character"))
-test_output_contains("class(my_logical)",
-                     incorrect_msg = sprintf(patt, "my_logical"))
-success_msg("Congratulations! This was the last exercise for this chapter. Head over to the next chapter to get immersed in the world of vectors!")
-```
 
+numeric_incorrect <- paste("&#3629;&#3618;&#3656;&#3634;&#3621;&#3610;&#3650;&#3588;&#3657;&#3604;&#3607;&#3637;&#3656;&#3626;&#3633;&#3656;&#3591;&#3649;&#3626;&#3604;&#3591;&#3588;&#3656;&#3634;&#3611;&#3619;&#3632;&#3648;&#3616;&#3607;&#3586;&#3657;&#3629;&#3617;&#3641;&#3621;&#3607;&#3637;&#3656;&#3629;&#3618;&#3641;&#3656;&#3651;&#3609;&#3605;&#3633;&#3623;&#3649;&#3611;&#3619;", "`my_numeric`")
+# อย่าลบโค้ดที่สั่งแสดงค่าประเภทข้อมูลที่อยู่ในตัวแปร `my_numeric`
 
+patt <- paste("&#3605;&#3619;&#3623;&#3592;&#3626;&#3629;&#3610;&#3651;&#3627;&#3657;&#3649;&#3609;&#3656;&#3651;&#3592;&#3623;&#3656;&#3634;&#3588;&#3640;&#3603;&#3652;&#3604;&#3657;&#3614;&#3636;&#3617;&#3614;&#3660;", "`class(%1$s)`", "&#3621;&#3591;&#3652;&#3611;&#3648;&#3614;&#3639;&#3656;&#3629;&#3626;&#3633;&#3656;&#3591;&#3649;&#3626;&#3604;&#3591;&#3611;&#3619;&#3632;&#3648;&#3616;&#3607;&#3586;&#3629;&#3591;&#3586;&#3657;&#3629;&#3617;&#3641;&#3621;&#3607;&#3637;&#3656;&#3629;&#3618;&#3641;&#3656;&#3651;&#3609;&#3605;&#3633;&#3623;&#3649;&#3611;&#3619;", "`%1$s`", "&#3648;&#3619;&#3637;&#3618;&#3610;&#3619;&#3657;&#3629;&#3618;&#3649;&#3621;&#3657;&#3623;")
+# ตรวจสอบให้แน่ใจว่าคุณได้พิมพ์ `class(%1$s)` ลงไปเพื่อสั่งแสดงประเภทของข้อมูลที่อยู่ใน `%1$s` เรียบร้อยแล้ว
+
+test_output_contains("class(my_numeric)", incorrect_msg = numeric_incorrect)
+test_output_contains("class(my_character)", incorrect_msg = sprintf(patt, "my_character"))
```

Comment:
I'm a bit afraid that this string replacement might not work properly given we have so many weird characters in the string :P  You can keep it as-is for now. I'll let you know if it doesn't work.